File: flutter_frontend/lib/models/department.dart
class Department {
  final int id;
  final int organizationId;
  final String? organizationName;
  final String departmentName;
  final int? parentDepartmentId;
  final String? parentDepartmentName;

  Department({
    required this.id,
    required this.organizationId,
    this.organizationName,
    required this.departmentName,
    this.parentDepartmentId,
    this.parentDepartmentName,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'] ?? 0,
      organizationId: json['organizationId'] ?? 0,
      organizationName: json['organizationName'],
      departmentName: json['departmentName'] ?? '',
      parentDepartmentId: json['parentDepartmentId'],
      parentDepartmentName: json['parentDepartmentName'] ?? json['parent_department_name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'organizationId': organizationId,
      'organizationName': organizationName,
      'departmentName': departmentName,
      'parentDepartmentId': parentDepartmentId,
    };
  }
}

File: backend - Copy/Models/Department.cs
using System;

namespace backend.Models
{
    public class Department
    {
        public int Id { get; set; }
        public int OrganizationId { get; set; }
        public string? OrganizationName { get; set; }
        public string? DepartmentName { get; set; }
        public int? ParentDepartmentId { get; set; }
        public string? ParentDepartmentName { get; set; }
        
        // Aliases for Dapper mapping from stored procedures
        public string? department_name { get => DepartmentName; set => DepartmentName = value; }
        public string? organization_name { get => OrganizationName; set => OrganizationName = value; }
        public string? parent_department_name { get => ParentDepartmentName; set => ParentDepartmentName = value; }
        public string? department { get => DepartmentName; set => DepartmentName = value; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }

    public class CreateDepartmentRequest
    {
        public int OrganizationId { get; set; }
        public string? OrganizationName { get; set; }
        public string? DepartmentName { get; set; }
        public int? ParentDepartmentId { get; set; }
    }
}
